migrations/2020_11_23_000000_add_nickname_column.php

```diff
@@ -7,7 +7,7 @@
     'up' => function(Builder $schema) {
         if (!$schema->hasColumn('users', 'nickname')) {
             $schema->table('users', function (Blueprint $table) use ($schema) {
-                $table->string('nickname', 150)->after('username')->index();
+                $table->string('nickname', 150)->after('username')->index()->nullable();
```

Comment:
I don't like editing old migrations, why not stick to the currently newly added migration ?

Reply:
When a migration is rolled back, it should reflect the state before the migration was applied. This isn't possible here, because we can't make a column non nullable if there are already null values. So the only way to retain this property is to make the state before the migration is applied also have the null value.